File: common_db_mongo/src/main/java/com/goexp/common/db/mongo/DBOperatorTemplate.java
package com.goexp.common.db.mongo;

import com.mongodb.client.MongoCollection;
import org.bson.Document;

import java.util.Objects;
import java.util.function.Consumer;

public class DBOperatorTemplate extends AbstractDBTemplate {

    public DBOperatorTemplate(String dbName, String tableName) {
        super(dbName, tableName);
        Objects.requireNonNull(dbName);
        Objects.requireNonNull(tableName);
    }

    public void exec(Consumer<MongoCollection<Document>> back) {

        Objects.requireNonNull(back);

        var db = mongoClient.getDatabase(dbName);
        back.accept(db.getCollection(tableName));

    }

}

File: common_db_mongo/src/main/scala/com/goexp/common/db/mongo/DBOperatorTemplate.scala
package com.goexp.common.db.mongo

import java.util.Objects

import com.mongodb.client.MongoCollection
import org.bson.Document

class DBOperatorTemplate(dbName: String,
                         tableName: String) extends AbstractDBTemplate(dbName, tableName) {
  Objects.requireNonNull(dbName)
  Objects.requireNonNull(tableName)

  def exec(callback: MongoCollection[Document] => Unit): Unit = {
    Objects.requireNonNull(callback)

    val db = mongoClient.getDatabase(dbName)
    callback(db.getCollection(tableName))
  }
}